mausignald/types.py

```diff
@@ -397,7 +419,7 @@ class MessageData(SerializableAttrs):
     contacts: List[SharedContact] = field(factory=lambda: [])
 
     group: Optional[Group] = None
-    group_v2: Optional[GroupV2ID] = field(default=None, json="groupV2")
+    group_v2: Optional[GroupV2] = field(default=None, json="groupV2")
```

Comment:
Do messages ever contain full group info? If not, a separate class like before (+ group_change) probably makes more sense than checking for `.title`. If they do, then maybe just wrap the title check in a `has_full_info` property or something like that.

Reply:
the GroupV2 that comes with a message contains GroupChange. I could move GroupChange to GroupV2ID, it just didn't seem logical to have a type called "ID" that contains more than just the most basic information, it also seemed to make more sense to not split up what is a single type in signald into 2 types in mautrix-signal, depending on what information we expect it to contain. It seemed better to check whether it contains the desired information (just checking for title seems sufficient, but on a second thought we might have to be more thorough than that).

If you still prefer to keep GroupV2ID and move GroupChange into it, I'm happy to revert the changes related to it.

Reply:
reverted